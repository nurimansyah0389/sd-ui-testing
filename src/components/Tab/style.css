.Tab {
    margin-top: 20px;
    padding-left: 120px;
}
.Tab__Section {
    display: flex;
}
.Tab__Section > div {
    color: var(--color-primary);
    cursor: pointer;
    padding-bottom: 30px;
}
.Tab__Section > div:hover {
    color: var(--color-lighten);
}
.Tab__Section > div:nth-child(1) {
    margin-right: 50px;
}
.Tab__Section > .selected {
    border-bottom: 1px solid var(--color-primary);
}
.Tab__Content {
    margin-left: -120px;
    border-top: 1px solid #ddd;
    padding: 10px 0;
    display: none;
}
.Tab__Content.show {
    display: block;
}
.Tab__Content > img {
    max-width: 100%;
}