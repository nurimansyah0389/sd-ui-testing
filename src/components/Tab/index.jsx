import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './style.css';

class TabComponent extends Component {

    static propTypes = {
        className: PropTypes.string,
        data: PropTypes.shape({
            flightDetail: PropTypes.string,
            priceDetail: PropTypes.string,
        })
    };

    state = {
        isOpened: ''
    };

    handleToggleCollapse = name => {
        // Check the state first
        const { isOpened } = this.state;
        if (isOpened === '') {
            this.setState({ isOpened: name });
        } else if (name === isOpened) {
            // Handle Collapse
            this.setState({ isOpened: '' });
        } else {
            // Set isOpened to others
            this.setState({ isOpened: name });
        }
    };

    render = () => {
        const { isOpened } = this.state;
        const { className, data } = this.props;
        return (
            <nav className={`Tab ${className ? className : ''}`}>
                <div className="Tab__Section">
                    <div className={isOpened === 'flightDetail' ? 'selected' : ''} onClick={() => this.handleToggleCollapse('flightDetail')}>Flight Detail</div>
                    <div className={isOpened === 'priceDetail' ? 'selected' : ''} onClick={() => this.handleToggleCollapse('priceDetail')}>Price Detail</div>
                </div>

                <div className={`Tab__Content ${isOpened ? 'show' : ''}`}>
                    {isOpened === 'flightDetail' && <img src={data.flightDetail} alt="" />}
                    {isOpened === 'priceDetail' && <img src={data.priceDetail} alt="" />}
                </div>
            </nav>
        );
    };
}
export default TabComponent;