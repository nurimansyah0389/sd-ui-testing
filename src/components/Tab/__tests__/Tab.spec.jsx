import React from 'react';
import Renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import flightData from '../../../data/flights.json';
import Tab from '../';

describe('Tab Spec', () => {
    let component = null;

    beforeEach(() => {
        component = (
            <Tab data={flightData[0]}/>
        );
    });

    it('Should render without error', () => {
        const tree = Renderer.create(component);
        expect(tree.toJSON()).toMatchSnapshot();
    });
    it('Should have "data" props to pass to the detail', () => {
        const wrapper = mount(component);
        expect(wrapper.prop('data')).toBeDefined();
    });
    it('Should expand the tab detail if clicked on each section title and have "isOpened" is not same as previous state', () => {
        // Arrange
        const wrapper = shallow(component);

        // Act
        wrapper.find('.Tab__Section > div:first-child').simulate('click');

        // Assert
        expect(wrapper.find('.Tab__Content').hasClass('show')).toBe(true);
        expect(wrapper.instance().state.isOpened).toBe('flightDetail');
    });
    it('Should collapse the tab detail if clicked on each section title and have "isOpened" is same as previous state', () => {
        // Arrange
        const wrapper = shallow(component);

        // Act
        wrapper.find('.Tab__Section > div:first-child').simulate('click');
        wrapper.find('.Tab__Section > div:first-child').simulate('click');

        // Assert
        expect(wrapper.find('.Tab__Content').hasClass('show')).toBe(false);
        expect(wrapper.instance().state.isOpened).toBe('');
    });
});