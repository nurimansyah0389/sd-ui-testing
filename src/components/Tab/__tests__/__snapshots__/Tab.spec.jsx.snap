// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Spec Should render without error 1`] = `
<nav
  className="Tab "
>
  <div
    className="Tab__Section"
  >
    <div
      className=""
      onClick={[Function]}
    >
      Flight Detail
    </div>
    <div
      className=""
      onClick={[Function]}
    >
      Price Detail
    </div>
  </div>
  <div
    className="Tab__Content "
  />
</nav>
`;
